pkg/meta/version_service_test.go

```diff
@@ -317,3 +317,167 @@ func Test_inc_version_during_switch(t *testing.T) {
+	s.IncrementBucket(ctx, b2, s1)
+	s.IncrementBucket(ctx, b2, s2)
+
+	s.IncrementBucketACL(ctx, b1, s1)
+	s.IncrementBucketACL(ctx, b1, s2)
+	s.IncrementBucketACL(ctx, b2, s1)
+	s.IncrementBucketACL(ctx, b2, s2)
+
+	s.IncrementBucketTags(ctx, b1, s1)
+	s.IncrementBucketTags(ctx, b1, s2)
+	s.IncrementBucketTags(ctx, b2, s1)
+	s.IncrementBucketTags(ctx, b2, s2)
+
+	s.IncrementObj(ctx, o1, s1)
+	s.IncrementObj(ctx, o1, s2)
+	s.IncrementObj(ctx, o2, s1)
+	s.IncrementObj(ctx, o2, s2)
+	s.IncrementObj(ctx, o3, s1)
+	s.IncrementObj(ctx, o3, s2)
+	s.IncrementObj(ctx, o4, s1)
+	s.IncrementObj(ctx, o4, s2)
+
+	s.IncrementACL(ctx, o1, s1)
+	s.IncrementACL(ctx, o1, s2)
+	s.IncrementACL(ctx, o2, s1)
+	s.IncrementACL(ctx, o2, s2)
+	s.IncrementACL(ctx, o3, s1)
+	s.IncrementACL(ctx, o3, s2)
+	s.IncrementACL(ctx, o4, s1)
+	s.IncrementACL(ctx, o4, s2)
+
+	s.IncrementTags(ctx, o1, s1)
+	s.IncrementTags(ctx, o1, s2)
+	s.IncrementTags(ctx, o2, s1)
+	s.IncrementTags(ctx, o2, s2)
+	s.IncrementTags(ctx, o3, s1)
+	s.IncrementTags(ctx, o3, s2)
+	s.IncrementTags(ctx, o4, s1)
+	s.IncrementTags(ctx, o4, s2)
+
+	s.DeleteBucketMeta(ctx, s1, b1)
+
+	ver, err := s.GetObj(ctx, o1)
+	r.NoError(err)
+	r.Zero(ver[s1])
+	r.EqualValues(2, ver[s2])
+
+	ver, err = s.GetObj(ctx, o2)
+	r.NoError(err)
+	r.Zero(ver[s1])
+	r.EqualValues(2, ver[s2])
+
+	ver, err = s.GetObj(ctx, o3)
+	r.NoError(err)
+	r.EqualValues(1, ver[s1])
+	r.EqualValues(2, ver[s2])
+
+	ver, err = s.GetObj(ctx, o4)
+	r.NoError(err)
+	r.EqualValues(1, ver[s1])
+	r.EqualValues(2, ver[s2])
+
+	ver, err = s.GetACL(ctx, o1)
+	r.NoError(err)
+	r.Zero(ver[s1])
+	r.EqualValues(2, ver[s2])
+
+	ver, err = s.GetACL(ctx, o2)
+	r.NoError(err)
+	r.Zero(ver[s1])
+	r.EqualValues(2, ver[s2])
+
+	ver, err = s.GetACL(ctx, o3)
+	r.NoError(err)
+	r.EqualValues(1, ver[s1])
+	r.EqualValues(2, ver[s2])
+
+	ver, err = s.GetACL(ctx, o4)
+	r.NoError(err)
+	r.EqualValues(1, ver[s1])
+	r.EqualValues(2, ver[s2])
+
+	ver, err = s.GetTags(ctx, o1)
+	r.NoError(err)
+	r.Zero(ver[s1])
+	r.EqualValues(2, ver[s2])
+
+	ver, err = s.GetTags(ctx, o2)
+	r.NoError(err)
+	r.Zero(ver[s1])
+	r.EqualValues(2, ver[s2])
+
+	ver, err = s.GetTags(ctx, o3)
+	r.NoError(err)
+	r.EqualValues(1, ver[s1])
+	r.EqualValues(2, ver[s2])
+
+	ver, err = s.GetTags(ctx, o4)
+	r.NoError(err)
+	r.EqualValues(1, ver[s1])
+	r.EqualValues(2, ver[s2])
+
+	ver, err = s.GetBucket(ctx, b1)
+	r.NoError(err)
+	r.Zero(ver[s1])
+	r.EqualValues(2, ver[s2])
+
+	ver, err = s.GetBucket(ctx, b2)
+	r.NoError(err)
+	r.EqualValues(1, ver[s1])
+	r.EqualValues(2, ver[s2])
+
+	ver, err = s.GetBucketACL(ctx, b1)
+	r.NoError(err)
+	r.Zero(ver[s1])
+	r.EqualValues(2, ver[s2])
+
+	ver, err = s.GetBucketACL(ctx, b2)
+	r.NoError(err)
+	r.EqualValues(1, ver[s1])
+	r.EqualValues(2, ver[s2])
+
+	ver, err = s.GetBucketTags(ctx, b1)
+	r.NoError(err)
+	r.Zero(ver[s1])
+	r.EqualValues(2, ver[s2])
+
+	ver, err = s.GetBucketTags(ctx, b2)
+	r.NoError(err)
+	r.EqualValues(1, ver[s1])
+	r.EqualValues(2, ver[s2])
```

Comment:
Would be nice to have some comments on this. 

E.g. why exactly these values are checked on delete bucket meta operation.

Reply:
i agree that test is very verbose. it sets all possible bucket and object version metadata for 2 buckets in 2 storages. then it calls `DeleteBucketMeta(bucket, storage)` and tests that that metadata only for given `<bucket, storage>` combination was removed and all other metadata is still there.